sale_block_cancellation/tests/test_sale_block_cancellation.py

```diff
@@ -0,0 +1,83 @@
+# coding: utf-8
+############################################################################
+#    Module Writen For Odoo, Open Source Management Solution
+#
+#    Copyright (c) 2016 Vauxoo - http://www.vauxoo.com
+#    All Rights Reserved.
+#    info Vauxoo (info@vauxoo.com)
+#    coded by: Luis Torres <luis_t@vauxoo.com>
+#    planned by: Sabrina Romero <sabrina@vauxoo.com>
+############################################################################
+from openerp.tests.common import TransactionCase
+from openerp.exceptions import ValidationError
+from openerp.tools.safe_eval import safe_eval
+
+
+class TestSaleBlockCancellation(TransactionCase):
+
+    def setUp(self):
+        super(TestSaleBlockCancellation, self).setUp()
+        self.wiz_obj = self.env['stock.transfer_details']
+        self.wiz_ret_obj = self.env['stock.return.picking']
+        self.return_picking_line = self.env['stock.return.picking.line']
+        self.proc_obj = self.env['procurement.order.compute.all']
+
+        self.sale_demo = self.env.ref('sale.sale_order_1')
+
+    def test_01_allow_cancel(self):
+        'Test that allow cancel Sale Order'
+        sale = self.sale_demo.copy({'order_policy': 'picking'})
+        sale.action_button_confirm()
+        sale.action_cancel_allow()
+        self.assertEquals(sale.state, 'cancel', 'Sale not cancelled')
+
+    def test_02_no_allow_cancel_by_state(self):
+        'Test that not allow cancel Sale Order, by pickings in done'
+        sale = self.sale_demo.copy({'order_policy': 'picking'})
+        sale.action_button_confirm()
+        for picking in sale.picking_ids:
+            if picking.state == 'confirmed':
+                picking.force_assign()
+                wiz = self.wiz_obj.with_context({
+                    'active_id': picking.id,
+                    'active_ids': [picking.id],
+                    'active_model': 'stock.picking'
+                }).create({'picking_id': picking.id})
+                wiz.do_detailed_transfer()
+        with self.assertRaisesRegexp(
+                ValidationError,
+                "All the pickings are in 'Ready to transfer'"):
+            sale.action_cancel_allow()
+        self.assertNotEquals(sale.state, 'cancel', 'Sale was cancelled')
+
+    def test_03_allow_cancel_correctly_returned(self):
+        'Test that allow cancel Sale Order, with pickings returned'
+        sale = self.sale_demo.copy({'order_policy': 'picking'})
+        sale.warehouse_id.write({
+            'delivery_steps': 'pick_pack_ship',
+        })
+        sale.action_button_confirm()
+        self.proc_obj.create({}).procure_calculation()
+        for picking in sale.picking_ids:
+            if picking.state == 'confirmed':
+                picking.force_assign()
```

Comment:
@luistorresm 

You must avoid to force assing,

https://github.com/Vauxoo/lodigroup/pull/451